### PR TITLE
add `--format` flag to `docker info`

### PR DESCRIPTION
**Pull request by AkihiroSuda**
**\- What I did**
add `--format` flag to `docker info`

**\- How to verify it**
`docker info --format {{json .}}`

Signed-off-by: Akihiro Suda suda.akihiro@lab.ntt.co.jp

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
CI failure seems unrelated to my change

**Comment by tonistiigi** [comment]
SGTM, changing to code-review

**Comment by AkihiroSuda** [comment]
rebased to upstream

**Comment by thaJeztah** [comment]
retest this please

File: api/client/system/info.go [hunk 1]
```diff
@@ -214,3 +232,14 @@ func runInfo(dockerCli *client.DockerCli) error {
 	}
 	return nil
 }
+
+func formatInfo(dockerCli *client.DockerCli, info types.Info, format string) error {
+	tmpl, err := templates.Parse(format)
+	if err != nil {
+		return cli.StatusError{StatusCode: 64,
```
Comment (vdemeester) [inline api/client/system/info.go]: Naive question : why status code 64 ? :angel:

**Comment by vdemeester** [comment]
LGTM :cat:

**Comment by thaJeztah** [comment]
LGTM

but interested about the status code 64 as well :smile:

**Comment by thaJeztah** [comment]
docs LGTM, but can you update the man page as well? https://github.com/docker/docker/blob/master/man/docker-info.1.md

**Comment by thaJeztah** [comment]
oh, perhaps you can add this flag to the completion scripts as well;
https://github.com/docker/docker/blob/master/contrib/completion/bash/docker
https://github.com/docker/docker/blob/master/contrib/completion/zsh/_docker

**Comment by AkihiroSuda** [comment]
The status code is just for consistency to this, but I'm not sure how 64 was determined 😓 : https: ://github.com/docker/docker/blob/849807d89a349db0f869a37c393203d812109d58/api/client/system/version.go#L70

I'll update PR after my summer vacation 😃

**Comment by thaJeztah** [comment]
@AkihiroSuda we can carry the PR if you prefer; it's just the docs and completion scripts

File: api/client/system/info.go (hunk 1 above)
Comment (AkihiroSuda) [inline api/client/system/info.go]: It is just copied from here :sweat_smile: https://github.com/docker/docker/pull/14194/files#diff-1d801b8b2dcffcde77b5995194bb3171R56

**Comment by AkihiroSuda** [comment]
done by myself, sorry for being late

**Comment by vdemeester** [comment]
docs LGTM :cat: 
/cc @thaJeztah

File: contrib/completion/fish/docker.fish [hunk 2]
```diff
@@ -393,6 +395,8 @@ complete -c docker -A -f -n '__fish_seen_subcommand_from unpause' -a '(__fish_pr
 
 # version
 complete -c docker -f -n '__fish_docker_no_subcommand' -a version -d 'Show the Docker version information'
+complete -c docker -A -f -n '__fish_seen_subcommand_from version' -s f -l format  -d 'Format the output using the given go template'
+complete -c docker -A -f -n '__fish_seen_subcommand_from version' -l help -d 'Print usage'
```
Comment (thaJeztah) [inline contrib/completion/fish/docker.fish]: :heart: even the fish completions!

**Comment by thaJeztah** [comment]
LGTM, awesome!

**Pull request merged**